mint-auditor/src/gnosis/eth_data_types.rs

```diff
@@ -3,16 +3,26 @@
 //! Type wrappers for Ethereum addresses.
 
 use super::Error;
+use mc_util_from_random::{CryptoRng, FromRandom, RngCore};
 use serde_with::{DeserializeFromStr, SerializeDisplay};
-use std::{fmt, str::FromStr};
+use std::{
+    fmt,
+    hash::{Hash, Hasher},
+    str::FromStr,
+};
 
 /// Ethereum 20 byte address.
 /// We currently do not store the decoded bytes since we want to maintain the
 /// original capitalization (which is how Ethereum addresses represent a
 /// checksum). We don't have a need for the raw bytes.
-#[derive(Clone, Debug, Default, DeserializeFromStr, SerializeDisplay)]
+#[derive(Clone, Debug, Default, DeserializeFromStr, Ord, PartialOrd, SerializeDisplay)]
```

Comment:
IIUC we'll need to implement `[Partial]Ord` with lowercasing, similar to `[Partial]Eq`. 

The [`PartialOrd` docs](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html) state:
> The methods of this trait must be consistent with each other and with those of [`PartialEq`](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html). The following conditions must hold:
> `a == b` if and only if `partial_cmp(a, b) == Some(Equal)`.

Reply:
Good catch